sdk-base/src/main/java/com/mapbox/maps/plugin/MapProjection.kt

```diff
@@ -0,0 +1,94 @@
+   */
+  object Mercator : MapProjection(MERCATOR_PROJECTION_NAME)
+
+  /**
+   * Globe projection is a custom map projection model for rendering the map wrapped around a full 3D globe.
+   * Conceptually it is the undistorted and unskewed “ground truth” view of the map
+   * that preserves true proportions between different areas of the map.
+   *
+   * Some layers are not supported when map is in Globe projection:
+   *  - circle
+   *  - custom
+   *  - fill extrusion
+   *  - heatmap
+   *  - model
+   *  - location indicator
+   *
+   * If Globe projection is set it will be switched automatically to Mercator projection
+   * when passing [TRANSITION_ZOOM_LEVEL] during zooming in.
+   *
+   * @see [TRANSITION_ZOOM_LEVEL] for more details what projection will be used depending on current zoom level.
+   */
+  object Globe : MapProjection(GLOBE_PROJECTION_NAME)
+
+  fun toValue() = Value.valueOf(hashMapOf(NAME_KEY to Value(projectionName)))
+
+  /**
+   * Custom toString method.
+   */
+  override fun toString() = when (this) {
+    Globe -> "Globe"
+    Mercator -> "Mercator"
+  }
+
+  /**
+   * Static methods and variables.
+   */
+  companion object {
+    /**
+     * Zoom level threshold where MapboxMap will automatically switch projection
+     * from [MapProjection.Mercator] to [MapProjection.Globe] or vice-versa
+     * if [MapProjectionDelegate.setMapProjection] was configured to use [MapProjection.Globe].
+     *
+     * If MapboxMap is using [MapProjection.Globe] and current map zoom level is >= [TRANSITION_ZOOM_LEVEL] -
+     * map will use [MapProjection.Mercator] and [MapProjectionDelegate.getMapProjection] will return [MapProjection.Mercator].
+     *
+     * If MapboxMap is using [MapProjection.Globe] and current map zoom level is < [TRANSITION_ZOOM_LEVEL] -
+     * map will use [MapProjection.Globe] and [MapProjectionDelegate.getMapProjection] will return [MapProjection.Globe].
+     *
+     * If MapboxMap is using [MapProjection.Mercator] - map will use [MapProjection.Mercator] for any zoom level and
+     * [MapProjectionDelegate.getMapProjection] will return [MapProjection.Mercator].
+     */
+    const val TRANSITION_ZOOM_LEVEL = 5.0
+
+    /**
+     * Constructs [MapProjection] from [Value].
+     *
+     * @throws [RuntimeException] if given [value] could not be casted to [MapProjection].
+     */
+    fun fromValue(value: Value): MapProjection {
+      @Suppress("UNCHECKED_CAST")
+      val map = value.contents as? HashMap<String, Value>
+      return when (map?.get(NAME_KEY)?.contents as? String) {
+        MERCATOR_PROJECTION_NAME -> Mercator
+        GLOBE_PROJECTION_NAME -> Globe
+        else -> throw RuntimeException("Could not cast given Value to valid MapProjection!")
+      }
+    }
+
+    private const val NAME_KEY = "name"
+    private const val MERCATOR_PROJECTION_NAME = "mercator"
+    private const val GLOBE_PROJECTION_NAME = "globe"
```

Comment:
would it better to generate these enums from gl-native?

Reply:
to leave API flexible it was decided to operate with pure `Value` type, it could have looked nicer if bindgen enum could be wrapped in a `Value` but if I recall correctly it's not possible.